LinkedList/LinkedList/TSLList.h

```diff
@@ -0,0 +1,352 @@
+#include <functional>
+
+template <typename Type>
+class TSLList
+{
+private:
+	int Size;
+	struct FNode
+	{
+		Type Element;
+		FNode* Next = nullptr;
+	};
+	FNode* Head = nullptr;
+	FNode* Tail = nullptr;
+
+public:
+	TSLList()
+	{
+		Size = 0;
+	}
+
+	~TSLList()
+	{
+		Clear();
+	}
+
+	//--------------------------------FIterator
+	class FIterator
+	{
+	private:
+		FNode* Current;
+
+	public:
+		FIterator(FNode* NewCurrent)
+		{
+			Current = NewCurrent;
+		}
+
+		FIterator& operator++()
+		{
+			Current = Current->Next;
+			return *this;
+		}
+
+		Type& operator*()
+		{
+			return Current->Element;
+		}
+
+		const Type& operator*() const
+		{
+			return Current->Element;
+		}
+
+		bool operator!=(const FIterator& Other)
+		{
+			return Current != Other.Current;
+		}
+
+		bool operator!=(const FIterator& Other) const
+		{
+			return Current != Other.Current;
+		}
+	};
+
+	FIterator begin()
+	{
+		if(Head)
+			return FIterator(Head);
+		return FIterator(nullptr);
+	}
+
+	FIterator end()
+	{
+		return FIterator(nullptr);
+	}
+
+	//-------------------------------CopyConstructor
+	TSLList(const TSLList<Type>& OtherTSLlist)
+	{
+		FNode* Temporal = OtherTSLlist.Head;
+
+		for (int i = 0; i < Size; i++)
+		{
+			AddTail(Temporal->Element);
+			Temporal = Temporal->Next;
+		}
+	}
+
+	//-------------------------Access
+	Type& operator[] (const int Index)
+	{
+		FNode* Temporal = Head; 
+		for (int i = 0; i < Index; i++)
+		{
+			Temporal = Temporal->Next;
+		}
+		return Temporal->Element;
+	}
+
+	const Type& operator[] (const int Index) const
+	{
+		FNode* Temporal = Head;
+		for (int i = 0; i < Index; i++)
+		{
+			Temporal = Temporal->Next;
+		}
+		return Temporal->Element;
+	}
+
+	//-----------------------------------------------
+	TSLList& operator= (const TSLList<Type>& OtherTSLlist)
+	{
+		Clear();
+		FNode* Temporal = OtherTSLlist.Head;
+
+		for (int i = 0; i < Size; i++)
+		{
+			AddTail(Temporal->Element);
+			Temporal = Temporal->Next;
+		}
+		return *this;
+
+	}
+
+	//-----------------------------------------------
+	Type& GetHead()
+	{
+		return Head->Element;
+	}
+
+	const Type& GetHead() const
+	{
+		return Head->Element;
+	}
+
+	//-----------------------------------------------
+	Type& GetTail()
+	{
+		return Tail->Element;
+	}
+
+	const Type& GetTail() const
+	{
+		return Tail->Element;
+	}
+
+	//---------------------------------Size
+	bool IsEmpty()
+	{
+		return Size == 0;
+	}
+
+	//------------------------------------------------
+	int GetSize()
+	{
+		return Size;
+	}
+
+	//-----------------------------------Modifiers
+	void AddHead(const Type& InHead)
+	{
+		FNode* NewHead = new FNode;
+		NewHead->Element = InHead;
+
+		if (Head == nullptr)
+		{
+			Head = NewHead;
+			Tail = NewHead;
+			Size++;
+		}
+		else
+		{
+			NewHead->Next = Head;
+			Head = NewHead;
+			Size++;
+		}
+	}
+
+	//-------------------------------------------------
+	void AddTail(const Type& InTail)
+	{
+
+		if (Head == nullptr)
+		{
+			AddHead(InTail);
+		}
+
```

Comment:
Very nitpicky but this space between the if and the else statement might be a bit confusing at first sight. This would be slightly more readable by removing line 189.